hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -581,6 +583,12 @@ public void initialize(URI name, Configuration originalConf)
 
       s3aInternals = createS3AInternals();
 
+      s3ObjectStorageClassFilter = Optional.ofNullable(conf.get(READ_RESTORED_GLACIER_OBJECTS))
```

Comment:
would prefer `conf.get(READ_RESTORED_GLACIER_OBJECTS, READ_ALL)` , meaning READ_ALL is the default. and then you can get rid of the `orElse()`

Reply:
or we just go for "upper case is required" and use what you've proposed. more brittle but simpler?